packages/insomnia-app/app/ui/components/wrapper-home.js

```
@@ -161,32 +163,59 @@ class WrapperHome extends React.PureComponent<Props, State> {
-        const fsPlugin = MemPlugin.createPlugin();
-        plugins.set('fs', fsPlugin);
+        const fs = MemPlugin.createPlugin();
+
+        repoSettingsPatch.uri = translateSSHtoHTTP(repoSettingsPatch.uri);
 
         // Pull settings returned from dialog and shallow-clone the repo
-        const { credentials, uri: url } = repoSettingsPatch;
+        const {
+          credentials: { username, token: password },
+          uri: url,
+        } = repoSettingsPatch;
+        console.log({ repoSettingsPatch });
+        const cloneParams = {
+          fs,
+          http,
+          core,
+          dir: GIT_CLONE_DIR,
+          gitdir: GIT_INTERNAL_DIR,
+          singleBranch: true,
+          url,
+          onAuth: () => ({
+            username,
+            password,
+          }),
+          depth: 1,
+        };
         try {
-          await git.clone({
-            core,
-            dir: GIT_CLONE_DIR,
-            gitdir: GIT_INTERNAL_DIR,
-            singleBranch: true,
-            url,
-            ...credentials,
-            depth: 1,
-            noGitSuffix: true,
-          });
+          // TODO: see if there's a way to retry
+          await git.clone(cloneParams);
         } catch (err) {
+          if (!cloneParams.url.endsWith('.git')) {
+            try {
+              await git.clone({
+                ...cloneParams,
+                url: addDotGit(cloneParams),
+              });
+
+              // by this point the clone was successful, so update with this syntax
+              repoSettingsPatch.uri = url;
+            } catch (error) {
+              // TODO: verify there's not a way to combine these
+              showError({
+                title: 'Error Cloning Repository with .git',
+                message: error.message,
+                error,
+              });
+            }
+          }
+
           showError({ title: 'Error Cloning Repository', message: err.message, error: err });
```

This flow is going to show two errors, right 🤔

---

From my testing it only shows the `Error Cloning Repository` error, never the other one, so that gets swallowed, which doesn't seem right, but I might be missing something.

I think the flow should be as follows:
1. If we fail to clone a repo, try with .git appended to the URL
2. If the appended .git URL clone also fails, show a single modal with the errors from *both* attempts, as otherwise we might cover up eg an initial 401 from the first one, and a 404 for the second one, in the case of a repo URL that should not have `.git`.

---

Don't think you're missing anything. It was either two errors or one gets swallowed because the `showError` utility always shows the same instance of the `ErrorModal`.

I think we should actually only show the first error. The second attempt (with the .git suffix) is something attempting internally, transparent to the user.

If the .git suffix succeeds, we update the URL (as per a previous convo), and if it fails, the user doesn't need to be prompted about it because it's nothing something they did (but maybe we should still log into console, as well as a `console.log('[git] attempting clone with .git suffix')` message).

---

Even tho it is something we do transparently, I don't think showing just the first (or the second) error is ideal, as that would cover up valid errors in certain cases, such as where you get a 404 for a non .git URL, and then a 401/403 on the correct URL when .git was automatically appended, because your token is invalid.

Since there are so many potential edge cases, I'd prefer exposing both errors to the user rather than having them need to go look in the logs/console.

---

> Showing just the second error would cause problems in the scenario mentioned in my previous comment, and showing just the first error would be a problem if for example you get a 404 for a non .git URL, and then a 401/403 on the correct URL when .git was automatically appended, because your token is invalid, as getting an error saying 404 is not helpful for debugging that.

Showing both sounds good, we do capture both, I'm sure we can come up with a way to word both into a single sentence.

Something like:

> Error cloning repository, attempts with `.git` suffix and without `.git suffix ended in failure:

---

> if for example you get a 404 for a non .git URL, and then a 401/403 on the correct URL when .git was automatically appended, because your token is invalid

Ah okay, yes, this is a totally valid situation and I agree.

In that case we can probably skip sending a stack trace to `showError` (because we have two stacks) but combining the messages would be the play. 👏🏽

---

lemme know if c944f893f80abd93899c425365f4546561a1922e is what you all have in mind.

---

@dimitropoulos Almost, but not quite, it needs to be something like this:
```
try {
  // Clone the repo
} catch (err) {
  // If the repo URL does not end with .git, try cloning it with .git
  if (!endsWith('.git')) {
    try {
      // The clone was successful, update the URL so that it's the new one that ends in .git
      repoSettingsPatch.uri = url;
      // Don't show an error, and don't return, as we need the flow to continue
    } catch (error) {
      // Show an error that combines the error messages from both the initial clone attempt, as well as the second one with the .git URL, and return.
      return false;
    }
  } else {
    // Show an error with the error message from the initial clone attempt, and return.
    return false; 
  }
}
```

---

ok, I implemented this logic in 6222a403bdb16fd7498cfdf4263c95cd9e74c78c

I realized that the return of this function is never used (later in the same function it returns undefined or returns void) so I updated that while I was here to be accurate (and consistent) which enabled reducing the branching (but the logic should be identical to what you wrote).